phpBB/phpbb/template/twig/lexer.php

```diff
@@ -118,7 +118,7 @@ public function tokenize($code, $filename = null)
 
 		// Replace all of our variables, {VARNAME}, with Twig style, {{ VARNAME }}
 		// Appends any filters
-		$code = preg_replace('#{([a-zA-Z0-9_\.]+)(\|[^}]+?)?}#', '{{ $1$2 }}', $code);
+		$code = preg_replace('#{([a-zA-Z_][a-zA-Z0-9_\.]*)(\|[^}]+?)?}#', '{{ $1$2 }}', $code);
```

Comment:
No, it won't match that.  Note the opening brace, that fulfills the same function as the caret you were proposing.  Anyhow, the caret would not work, because we are working in the lexer, and therefore the match must trigger anywhere in the string, not only at the beginning.  So, this regex matches `{test123}` but not `{123test}` and, the real issue, it won't match `{123}` either.